### DISCUSSION TITLE
Question about IN statement and empty array

### DISCUSSION THREAD

**Discussion by sdespont** [doctrine/doctrine2#10016] [Support Questions] [answered]
Consider this DQL

```
        return $this->getEntityManager()->createQuery(
            'SELECT rf FROM Accounting:ReminderFee rf WHERE rf.account_detail NOT IN (:accountDetailIdList)'
        )->setParameter('accountDetailIdList', $idArray)
            ->getResult();
```

Is there normal that in case of the `$idArray `is an empty array, the runnable query contains the statement `NOT IN (NULL)` ? Because at least with MySql, `IN (NULL)` statement returns no row at all.

**Comment by SenseException** [comment, marked as answer]
It's the current state for the behaviour of `IN` and `NOT IN`. There's an old discussion in one of the PRs from the past in #8035

**Reply by sdespont** [reply]
Wouah, that's a long story :o) In my point of view, the fact that an empty array is usually an error and should not happen, therefore must throw an error.

**Reply by SenseException** [reply]
The best is to check if `accountDetailIdList` is empty before you run your query in your use case.

**Reply by sdespont** [reply]
Yes, thank you for the answer. I will handle the case in my side as you suggest. The problem occurs whit `NOT IN` statement because `NOT IN (NULL)` returns no result instead of all results.
